docs/guide/java_serialization_guide.md

```diff
@@ -510,6 +510,10 @@ If you create fury without setting `CompatibleMode` to `org.apache.fury.config.C
 strange
 serialization error, it may be caused by class inconsistency between serialization peer and deserialization peer.
 
+If you create fury and set `CompatibleMode` to `org.apache.fury.config.CompatibleMode.COMPATIBLE`, you need to register
+the class that you want to serialize or deserialize(You don't need to register any class inside that class).
```

Comment:
The registration is not mandatory, it's necessary just because you want deserialize one class into another, so you use class registration to setup class mapping

Reply:
Could we add a special section to document how to deserialize one pojo into another pojo, and add this note there?